src/main/java/frc/robot/events/EventWatcherThread.java

```diff
@@ -0,0 +1,50 @@
+package frc.robot.events;
+
+import edu.wpi.first.wpilibj.command.Scheduler;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+
+/** Singleton thread for monitoring things. */
+public class EventWatcherThread extends Thread {
+
+  private ArrayList<Event> queue = new ArrayList<>();
+  private LinkedHashMap<Event, Boolean> lastStateCache = new LinkedHashMap<>();
+
```

Comment:
Do we need both data structures for Events?

Reply:
Nope, I wrote this in an hour last year and never had time to go back to it because that year was really busy.